fixed-price-subscriptions/server/ruby/config_helper.rb

```diff
@@ -0,0 +1,237 @@
+    @vars = Dotenv.parse(*args)
+  end
+
+  def set_dotenv!(key, value)
+    @vars[key] = value
+    write_dotenv!
+  end
+
+  def write_dotenv!
+    File.open('.env', 'w') do |f|
+      REQUIRED_VARS.each do |v|
+        f.puts "#{v}=#{@vars[v]}"
+      end
+    end
+  end
+
+  def valid_prices?
+    _valid_price?('BASIC', 1200) &&
+      _valid_price?('PREMIUM', 1800)
+  end
+
+  def _valid_price?(price_id, unit_amount)
+    price = ENV[price_id]
+    if price.nil? || price == '' || price.include?('...')
+      puts <<~DOC
+        Environment variable missing for #{price_id}. This should be the ID
+        of a Price object in your Stripe account. You can create a product and related prices in the dashbaord here: https://dashboard.stripe.com/test/products.
+        Would you like us to create a Price now? [Y/n]
+      DOC
+      c = gets.chomp
+      if c.upcase == "N"
+        puts "Please create a Product and Price from the dashboard, then set its ID in .env."
+        return false
+      else
+        stripe_price = Stripe::Price.create(
+          unit_amount: 1200,
+          currency: "USD",
+          recurring: {
+            interval: 'month',
+          },
+          product_data: {
+            name: 'Sample Price',
+            metadata: {
+              stripe_sample_id: 'subscription-use-cases',
+              stripe_sample_integration_id: 'fixed-price',
+              stripe_sample_lang: 'ruby',
+            }
+          }
+        )
+        puts <<~DOC
+          Price created!
+          #{stripe_price.to_json}
+        DOC
+        set_dotenv!(price_id, stripe_price.id)
+      end
+      return false
+    end
+    true
+  end
+
+  def valid_paths?
+    static_dir = ENV['STATIC_DIR']
+    if static_dir.nil?
+      puts <<~DOC
+        No STATIC_DIR environment variable found in `.env`. This should be the
+        relative path to the directory where the client side HTML code is.
+      DOC
+      return false
+    end
+    static_dir_path = File.join(File.dirname(__FILE__), static_dir)
+    if !File.directory?(static_dir_path)
+      puts <<~DOC
+        The path to the STATIC_DIR is not a directory. Please check .env.
+      DOC
+      return false
+    end
+
+    if !File.exists?(File.join(static_dir_path, 'index.html'))
+      if static_dir == ''
+        puts <<~DOC
+          No value set for STATIC_DIR. If this sample was installed with the
+          Stripe CLI then STATIC_DIR is usually `../client`. If this sample was
+          git cloned, STATIC_DIR is typically set to `../../client/{vanillajs|react}`.
+        DOC
+        return false
+      else
+        puts <<~DOC
+          No `index.html` file found in #{static_dir_path}. Please check #{File.join(static_dir_path, 'index.html')}
+        DOC
+      end
+      return false
+    end
+
+    true
+  end
+
+  def valid_api_keys?
+    sk = ENV['STRIPE_SECRET_KEY']
+    if sk.nil? || (!sk.start_with?('sk_test_') && !sk.start_with?('rk_test_'))
```

Comment:
Worth double checking: I think there are valid legacy API keys without the `_test_` string (possibly predating the `sk_` prefix?). I could be wrong but wanted to point out if this ring any bells for you too. I don't think we need to make any change as its reasonable to assume developers here will be ~newer accounts or otherwise should generate a new longer key.

Reply:
There isn't that's a valid point. I was trying to make sure (1) people to use test keys rather than live (2) they have an actual key set :)